### ISSUE TITLE
Question about a module that does not required grad

### ISSUE THREAD

**Issue by seanie12** [facebookresearch/higher#8]
Thanks for the wonderful library!

I have one question about how to use higher.

I want to train the network with second order derivative,

but some parts of network (such as nn.Embedding ) are frozen.

something like this
```
net= Net()
param = filter(lambda x: x.requires_grad, model.parameters())
inner_opt = torch.optim.SGD(param, lr=1e-1)

with higher.innerloop_ctx(net, inner_opt, copy_initial_weights=False) as (fnet, diffopt):
  logits = net(batch_x)
  loss = loss_fnt(logits, batch_y)
  diffopt.step(loss)
              
```
but i got the error message "RuntimeError: One of the differentiated Tensors does not require grad"

**Comment by egrefen** [comment]
Thanks for flagging this, @seanie12. This is probably a bug. Would you mind providing a minimal repro snippet (e.g. add the simplest definition of `Net`, imports, and `loss_fnt` def which raise the error)?

**Comment by seanie12** [comment]
Sure.! Here it is
```
import torch
import torch.nn as nn
import higher


class Net(nn.Module):
    def __init__(self):
        super(Net, self).__init__()
        self.fc1 = nn.Linear(30, 50)
        self.fc2 = nn.Linear(50, 1)
        # freeze first FC layer
        for param in self.fc1.parameters():
            param.requires_grad = False

    def forward(self, x):
        hidden = self.fc1(x)
        logits = self.fc2(hidden).squeeze(1)
        return logits


# random input and labels for debugging
inputs = torch.randn(16, 30)
ones = torch.ones(8)
zeros = torch.zeros(8)
labels = torch.cat([ones, zeros], dim=0)

net = Net()

param = filter(lambda x: x.requires_grad, net.parameters())
inner_opt = torch.optim.SGD(param, lr=1e-1)
loss_fnt = nn.BCEWithLogitsLoss()

with higher.innerloop_ctx(net, inner_opt, copy_initial_weights=False) as (fnet, diffopt):
    logits = net(inputs)
    loss = loss_fnt(logits, labels)
    diffopt.step(loss)
```

**Comment by egrefen** [comment]
Thanks @seanie12. I'll try to look into this by the end of the week.

**Comment by seanie12** [comment]
Any updates so far?

**Comment by egrefen** [comment]
On my to-do list for tomorrow!

**Comment by egrefen** [comment]
@seanie12 I fixed the line in your example where the outer net was incorrectly used instead of the patched net, and removed `copy_initial_weights=False` from the context manager since it shouldn't be relevant (unless it is). I have a fix which I'll push tonight.

**Comment by egrefen** [comment]
Fixed in #11. Please re-open if this does not address your issue.
